graph/stream-enact-events.go

```diff
@@ -207,16 +250,16 @@ func (gt *GraphThread[V, E, M, N]) checkInsertPending(sidx uint32, pending uint3
 }
 
 // Main function to enact topology events. Will look through the topology event buffer and apply changes to the graph.
-func EnactTopologyEvents[EP EPP[E], V VPI[V], E EPI[E], M MVI[M], N any, A Algorithm[V, E, M, N]](alg A, g *Graph[V, E, M, N], gt *GraphThread[V, E, M, N], changeCount uint64, delOnExpire uint64) (addEvents uint32, delEvents uint32) {
+func EnactTopologyEvents[EP EPP[E], V VPI[V], E EPI[E], M MVI[M], N any, A Algorithm[V, E, M, N]](alg A, g *Graph[V, E, M, N], gt *GraphThread[V, E, M, N], changeCount uint64, delOnExpire uint64, undirected bool) (addEvents uint32, delEvents uint32) {
 	uniqueCount := uint64(0)
 	canCheckTimestamp := (g.Options.TimestampPos != 0) || g.Options.LogicalTime
-	var ok bool
 	var sidx uint32
 
 	for i := uint64(0); i < changeCount; i++ {
-		if sidx, ok = gt.VertexMap[gt.TopologyEventBuff[i].SrcRaw]; !ok {
-			sidx = NewVertex(alg, g, gt, gt.TopologyEventBuff[i].SrcRaw, gt.TopologyEventBuff[i].EventIdx())
+		if !undirected && gt.TopologyEventBuff[i].EventIdx()%2 == 0 {
+			continue // Skip even events for directed graphs.
```

Comment:
On direct graphs, why do both src and dst still need to remit to each other? It seems like the remit events from src to dst are just ignored.